openmc/summary.py

```diff
@@ -0,0 +1,229 @@
+from collections.abc import Iterable
+import re
+import warnings
+
+import numpy as np
+import h5py
+
+import openmc
+import openmc.checkvalue as cv
+from openmc.region import Region
+
+_VERSION_SUMMARY = 6
+
+
+class Summary(object):
+    """Summary of geometry, materials, and tallies used in a simulation.
+
+    Attributes
+    ----------
+    date_and_time : str
+        Date and time when simulation began
+    geometry : openmc.Geometry
+        The geometry reconstructed from the summary file
+    materials : openmc.Materials
+        The materials reconstructed from the summary file
+    nuclides : dict
+        Dictionary whose keys are nuclide names and values are atomic weight
+        ratios.
+    macroscopics : list
+        Names of macroscopic data sets
+    version: tuple of int
+        Version of OpenMC
+
+    """
+
+    def __init__(self, filename):
+        if not filename.endswith(('.h5', '.hdf5')):
+            msg = 'Unable to open "{0}" which is not an HDF5 summary file'
+            raise ValueError(msg)
+
+        self._f = h5py.File(filename, 'r')
+        cv.check_filetype_version(self._f, 'summary', _VERSION_SUMMARY)
+
+        self._geometry = openmc.Geometry()
+
+        self._fast_materials = {}
+        self._fast_surfaces = {}
+        self._fast_cells = {}
+        self._fast_universes = {}
+        self._fast_lattices = {}
+
+        self._materials = openmc.Materials()
+        self._nuclides = {}
+        self._macroscopics = []
+
+        self._read_nuclides()
+        self._read_macroscopics()
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", openmc.IDWarning)
+            self._read_geometry()
+
+    @property
+    def date_and_time(self):
+        return self._f.attrs['date_and_time'].decode()
+
+    @property
+    def geometry(self):
+        return self._geometry
+
+    @property
+    def materials(self):
+        return self._materials
+
+    @property
+    def nuclides(self):
+        return self._nuclides
+
+    @property
+    def macroscopics(self):
+        return self._macroscopics
+
+    @property
+    def version(self):
+        return tuple(self._f.attrs['openmc_version'])
+
+    def _read_nuclides(self):
+        if 'nuclides/names' in self._f:
+            names = self._f['nuclides/names'][()]
+            awrs = self._f['nuclides/awrs'][()]
+            for name, awr in zip(names, awrs):
+                self._nuclides[name.decode()] = awr
+
+    def _read_macroscopics(self):
+        if 'macroscopics/names' in self._f:
+            names = self._f['macroscopics/names'][()]
+            for name in names:
+                self._macroscopics = name.decode()
```

Comment:
this just assigns value to `_macroscopics` multiple times? Are you missing a break here?